source/develop/release-management/features/virt/secure-cpus.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Default / Secure CPU Type Concept
+category: feature
+authors: ljelinko, mskrivan
+feature_name: Default / Secure CPU Type Concept
+feature_modules: engine
+---
+
+# Default / Secure CPU Type Concept
+
+## Summary
+
+Improve the maintainability of oVirt CPU Types by supporting only _default_ and _secure_ version of each CPU microarchitecture.
+
+## Owner
+
+*   Name: Lucia Jelinkova (ljelinko)
+
+*   Email: <ljelinko@redhat.com>
+
+## Detailed Description
+
+oVirt keeps a list of supported CPU Types in the vdc_options table in the database under the key ServerCPUList. Previously, with every security update a new CPU Type was created for ALL affected architectures. Consequently, the list grew in length and looked like this:
+
+- Intel Broadwell Client Family
+- Intel Broadwell Client IBRS Family'
+- Intel Broadwell Client IBRS SSBD Family'
+- Intel Broadwell Client IBRS SSBD MDS Family
+- Intel Skylake Client Family
+- Intel Skylake Client IBRS Family'
+- Intel Skylake Client IBRS SSBD Family'
+- Intel Skylake Client IBRS SSBD MDS Family
+- Intel Skylake Server Family
+- Intel Skylake Server IBRS Family'
+- Intel Skylake Server IBRS SSBD Family'
+- Intel Skylake Server IBRS SSBD MDS Family
+
+In order to keep the CPU list manageable, only 2 CPU Types are now supported for any CPU microarchitecture that has security updates. 
+
+- Intel Broadwell Client Family
+- Secure Intel Broadwell Client Family'
+- Intel Skylake Client Family
+- Secure Intel Skylake Client Family'
+- Intel Skylake Server Family
+- Secure Intel Skylake Server Family'
+
+The _default_ CPU Type will not change over the time while the _secure_ CPU type will contain the latest updates. 
+
+## Prerequisites
+Previously, it was not possible to simply update the CPU Type configuration because all hosts that would not comply to the new configuration would become non operational.
+
+To ensure that the hosts are operational even after the CPU configuration is changed, we store the currently used configuration in the cluster table. That ensures that even after configuration in vdc_options table is changed, the hosts stay operational. 
+
+When all cluster hosts are compatible with the changed CPU, the cluster table is automatically updated to contain the same configuration as vdc_options table. 
+
+## Benefit to oVirt
+
+Improved maintainability of supported CPU Types.
+
+## User Experience
+
+If the CPU configuration is different in the cluster table than in the vdc_options table, it is important to notify the user. The following warnings are displayed in the UI:
+
+- cluster list - a warning that the cluster contains host(s) that are not fully compatible with the current CPU Type configuration
+- host list and detail - a warning that the host is not fully compatible with the current CPU Type configuration
+- vm list and detail - a warning that the VM CPU configuration is not compatible with the current CPU Type configuration
+
```

Comment:
Can you also provide details of REST API changes?
ie- how do I know what is secure and what is default using the API? etc.

Reply:
This feature should not affect the REST API other then that CPU Types are renamed but I've stated it there explicitly.